packages/target-tauri/src-tauri/src/lib.rs

```diff
@@ -87,6 +90,8 @@ pub fn run() {
             }));
     }
 
+    builder = create_main_menu(builder);
+
```

Comment:
you don't need to define it here, you can do it later like this:

```rs
    window.set_menu(menu)
        // for macOS use this and have some switch logic to switch the menu based on focused window
        // (implemented as tauti plugin or probably also would work as state)
        app.set_menu(menu)
```

besides that there are quite some cases where it needs to be updated/rebuild:
- on MacOS on changing window
- when translations change (@nojisan will make port the translation function to rust, so we can also use it in the backend)
- maybe also  when some kind of values change?

Reply:
I still need the zoom level on startup to set the proper check in the zoom settings. When adding the menu later I firstly don't see where a good location for that would be and users could see bad state.

Reply:
try it out, I bet it is so fast that the other menu is not even noticed by anyone during startup. the location for it should be the place where delta core and so on is loaded: somewhere at the end of the setup function/closure: `.setup(move |app| {`